tests/call-handler.test.ts

```diff
@@ -1,144 +1,18 @@
 import { RPCHandler } from "../types/rpc-handler";
 import { HandlerConstructorConfig } from "../types/handler";
 import { PrettyLogs } from "../types/logs";
 
-export const testConfig: HandlerConstructorConfig = {
-  networkId: "100",
-  autoStorage: false,
-  cacheRefreshCycles: 3,
-  networkName: null,
-  networkRpcs: null,
-  rpcTimeout: 600,
-  runtimeRpcs: null,
-  proxySettings: {
-    retryCount: 3,
-    retryDelay: 10,
-    logTier: "info",
-    logger: new PrettyLogs(),
-    strictLogs: true,
-  },
-};
-
-const nonceBitmapData = {
-  to: "0x000000000022D473030F116dDEE9F6B43aC78BA3",
-  data: "0x4fe02b44000000000000000000000000d9530f3fbbea11bed01dc09e79318f2f20223716001fd097bcb5a1759ce02c0a671386a0bbbfa8216559e5855698a9d4de4cddea",
-};
-const rpcList = [
-  { url: "http://127.0.0.1:85451" },
-  { url: "http://127.0.0.1:85454" },
-  { url: "http://127.0.0.1:85453" },
-  { url: "http://127.0.0.1:854531" },
-  { url: "http://127.0.0.1:854532" },
-  { url: "http://127.0.0.1:854533" },
-  { url: "http://127.0.0.1:854535" },
-  { url: "http://127.0.0.1:854" },
-  { url: "http://127.0.0.1:85" },
-  { url: "http://127.0.0.1:81" },
-  { url: "http://127.0.0.1:8545" },
-];
-const txHashRegex = new RegExp("0x[0-9a-f]{64}");
+jest.mock("axios", () => ({
+  ...jest.requireActual("axios"),
+  create: jest.fn(() => ({
+    post: jest.fn(),
+  })),
+}));
 
 describe("Call Handler", () => {
-  afterAll(() => {
+  afterEach(() => {
     jest.clearAllMocks();
     jest.clearAllTimers();
-    jest.resetAllMocks();
-    jest.resetModules();
-  });
-
-  beforeEach(() => {
-    jest.clearAllMocks();
-    jest.clearAllTimers();
-    jest.resetAllMocks();
-    jest.resetModules();
-  });
-
-  describe("createProviderProxy", () => {
-    let provider: JsonRpcProvider;
-
-    it("should make a successful get_blockNumber call", async () => {
-      await jest.isolateModulesAsync(async () => {
-        const module = await import("../types/rpc-handler");
-        const handler = new module.RPCHandler({ ...testConfig, proxySettings: { ...testConfig.proxySettings, logTier: "verbose" } });
-
-        provider = await handler.getFastestRpcProvider();
-        const blockNumber = await provider.send("eth_blockNumber", []);
-        expect(parseInt(blockNumber)).toBeGreaterThan(0);
-      });
-    }, 15000);
-
-    it("should make a successful eth_call", async () => {
-      await jest.isolateModulesAsync(async () => {
-        const module = await import("../types/rpc-handler");
-        const handler = new module.RPCHandler({ ...testConfig, proxySettings: { ...testConfig.proxySettings, logTier: "verbose" } });
-
-        provider = await handler.getFastestRpcProvider();
-        const response = await provider.send("eth_call", [nonceBitmapData, "latest"]);
-        expect(response).toBeDefined();
-        expect(response).toBe("0x" + "00".repeat(32));
-      });
-    }, 15000);
-  });
-
-  describe("Write ops cases", () => {
```

Comment:
Why have we removed all of the happy path tests, seems like we are only testing for failure cases?

Reply:
Cf my comment on the main thread